services/cache.js

```diff
@@ -0,0 +1,22 @@
+export default class ForecastCache {
+	constructor({ apiClient, dbClient }) {
+		this.apiClient = apiClient;
+		this.dbClient = dbClient;
+	}
+
+	async getByZipAndTS(zip, ts) {
+		const cachedForecast = await this.dbClient.getByZipAndTS(zip, ts);
```

Comment:
`thought`: We should probably rename this method to something like `getRecordByZipcodeAndTimestampAndMaxTime`